### PR TITLE
Fix metadata local

### LINKED ISSUE #6: Add local file directories as data sources
It would be great to be able to read metadata from local file system so that we can perform tasks like: Give me all S2 scenes in directory x with less than y% cloud cover. [TinyDB](https://tinydb.readthedocs.io/en/latest/) could be a possible package to get started with.

### PR DESCRIPTION
**Pull request by MWieland**
Closes #6

### CODE REVIEW

File: ukis_pysat/data.py [hunk 1]
```diff
@@ -33,18 +34,18 @@ class Source:
     Remote APIs and local data directories that hold metadata files are supported.
     """
 
-    def __init__(self, source, source_dir=None):
+    def __init__(self, datahub, datadir=None):
```
Comment (fwfichtner) [inline ukis_pysat/data.py:37]: that's a lot less confusing

File: ukis_pysat/data.py [hunk 2]
```diff
@@ -33,18 +34,18 @@ class Source:
...
         """
-        self.src = source
+        self.src = datahub
 
-        if self.src == Datahub.file:
-            if not source_dir:
-                raise AttributeError(f"{traceback.format_exc()} 'source_dir' has to be set if source is {self.src}.")
+        if self.src == Datahub.File:
+            if not datadir:
+                raise AttributeError(f"{traceback.format_exc()} 'datadir' has to be set if datahub is {self.src}.")
```
Comment (fwfichtner) [inline ukis_pysat/data.py:46]:
```suggestion
                raise AttributeError(f"'datadir' has to be set if datahub is 'File'.")

```

File: ukis_pysat/data.py [hunk 3]
```diff
@@ -6,6 +6,7 @@
 import os
 import shutil
 import traceback
```
Comment (fwfichtner) [inline ukis_pysat/data.py:8]:
```suggestion

```

File: README.md [hunk 4]
```diff
@@ -30,7 +30,6 @@ Here's an example about some basic features, it might also help to read through
 
 ````python
 from ukis_pysat.data import Source
-from ukis_pysat.raster import Image
```
Comment (fwfichtner) [inline README.md:33]: this is still needed here, see line 65

File: ukis_pysat/members.py [hunk 5]
```diff
@@ -11,6 +11,6 @@ class Platform(Enum):
 
 
 class Datahub(Enum):
-    file = auto()
+    File = auto()
```
Comment (fwfichtner) [inline ukis_pysat/members.py:14]: is it worth breaking compatibility for a capital letter here? @BoehnkeC are you using this already?

File: ukis_pysat/data.py [hunk 6]
```diff
@@ -61,23 +62,74 @@ def __init__(self, source, source_dir=None):
...
+                            m_platform == platform.value
+                            and m_date >= start_date
+                            and m_date < end_date
+                            and m_geom.intersects(geom)
+                            and m_cloud_cover >= min_cloud_cover
+                            and m_cloud_cover < max_cloud_cover
+                        ):
+                            meta_src.append(m)
+                    except (json.decoder.JSONDecodeError, LookupError, TypeError) as e:
+                        warnings.warn(
```
Comment (fwfichtner) [inline ukis_pysat/data.py:128]: why do we not throw an error here and remove try... except? I think such things should be handled outside of the library.

File: ukis_pysat/members.py (hunk 5 above)
Comment (BoehnkeC) [reply ukis_pysat/members.py:14]: Haven't used it yet.

File: README.md (hunk 4 above)
Comment (MWieland) [reply README.md:33]: Sorry didnt see that one. Thanks for correcting.

File: ukis_pysat/members.py (hunk 5 above)
Comment (MWieland) [reply ukis_pysat/members.py:14]: I know its a bit picky, but better now than later.

File: ukis_pysat/data.py (hunk 6 above)
Comment (MWieland) [reply ukis_pysat/data.py:128]: this is a good point. then lets generally agree on moving such things outside of the library. will modify this tomorrow accordingly.

**Pull request merged**